src/olympia/files/utils.py

```diff
@@ -837,3 +842,89 @@ def _update_version_in_json_manifest(content, new_version_number):
     if 'version' in updated:
         updated['version'] = new_version_number
     return json.dumps(updated)
+
+
+def extract_translations(file_obj):
+    """Extract all translation messages from `file_obj`.
+
+    :param locale: if not `None` the list will be restricted only to `locale`.
+    """
+    if isinstance(file_obj, basestring):
+        xpi_path = file_obj
+    elif isinstance(file_obj, DjangoFile):
+        xpi_path = file_obj
+    elif hasattr(file_obj, 'path'):  # FileUpload
+        xpi_path = file_obj.path
+    elif hasattr(file_obj, 'name'):  # file-like object
+        xpi_path = file_obj.name
+    else:
+        raise ValueError('A path, file-like or FileUpload object is required')
+
+    messages = {}
+
+    try:
+        with zipfile.ZipFile(xpi_path, 'r') as source:
+            file_list = source.namelist()
+
+            locales = {name[8:].strip('/') for name in file_list
```

Comment:
Also, the whole thing is weird enough that I think you should add a comment. Maybe just linking https://developer.chrome.com/extensions/i18n is enough.